### ISSUE TITLE
Error polling connection (KeyError: 0)

### ISSUE THREAD

**Issue by fuerst** [jeffreywildman/homebrew-virt-manager#16]
I got `virt-manager` working once with successfully connecting to remote servers. Since then I always get the somewhat generic error `Error polling connection` regardless the server I'm using (tried 4). Any ideas where to look for or how to debug?

````
$ LANG=C virt-manager -c qemu+ssh://root@server/system?socket=/var/run/libvirt/libvirt-sock  --debug
[Tue, 09 Jun 2015 09:45:46 virt-manager 12526] DEBUG (cli:246) Launched with command line: /usr/local/Cellar/virt-manager/1.2.0/libexec/share/virt-manager/virt-manager -c qemu+ssh://root@server/system?socket=/var/run/libvirt/libvirt-sock --debug
[Tue, 09 Jun 2015 09:45:46 virt-manager 12526] DEBUG (virt-manager:153) virt-manager version: 1.2.0
[Tue, 09 Jun 2015 09:45:46 virt-manager 12526] DEBUG (virt-manager:154) virtManager import: <module 'virtManager' from '/usr/local/Cellar/virt-manager/1.2.0/libexec/share/virt-manager/virtManager/__init__.pyc'>
[Tue, 09 Jun 2015 09:45:46 virt-manager 12526] DEBUG (virt-manager:213) GTK version: 3.16.3
[Tue, 09 Jun 2015 09:45:46 virt-manager 12526] DEBUG (engine:456) libguestfs inspection support: False
[Tue, 09 Jun 2015 09:45:46 virt-manager 12526] DEBUG (systray:150) Showing systray: False
[Tue, 09 Jun 2015 09:45:46 virt-manager 12526] DEBUG (engine:228) About to connect to uris ['qemu+ssh://root@server/system?socket=/var/run/libvirt/libvirt-sock']
[Tue, 09 Jun 2015 09:45:46 virt-manager 12526] DEBUG (manager:207) Showing manager
[Tue, 09 Jun 2015 09:45:46 virt-manager 12526] DEBUG (engine:357) window counter incremented to 1
[Tue, 09 Jun 2015 09:45:46 virt-manager 12526] DEBUG (connection:541) conn=qemu+ssh://root@server/system?socket=/var/run/libvirt/libvirt-sock changed to state=Connecting
[Tue, 09 Jun 2015 09:45:46 virt-manager 12526] DEBUG (connection:845) Scheduling background open thread for qemu+ssh://root@server/system?socket=/var/run/libvirt/libvirt-sock
[Tue, 09 Jun 2015 09:45:47 virt-manager 12526] DEBUG (connection:894) libvirt version=1002016
[Tue, 09 Jun 2015 09:45:47 virt-manager 12526] DEBUG (connection:896) daemon version=9008
[Tue, 09 Jun 2015 09:45:47 virt-manager 12526] DEBUG (connection:897) conn version=1000000
[Tue, 09 Jun 2015 09:45:48 virt-manager 12526] DEBUG (connection:899) qemu+ssh://root@server/system?socket=/var/run/libvirt/libvirt-sock capabilities:
<capabilities>

  <host>
    <uuid>1e00ae60-008c-6200-f1e1-08606e45db68</uuid>
    <cpu>
      <arch>x86_64</arch>
      <model>Westmere</model>
      <vendor>Intel</vendor>
      <topology sockets="1" cores="4" threads="2"/>
      <feature name="rdtscp"/>
      <feature name="x2apic"/>
      <feature name="xtpr"/>
      <feature name="tm2"/>
      <feature name="est"/>
      <feature name="vmx"/>
      <feature name="ds_cpl"/>
      <feature name="monitor"/>
      <feature name="pbe"/>
      <feature name="tm"/>
      <feature name="ht"/>
      <feature name="ss"/>
      <feature name="acpi"/>
      <feature name="ds"/>
      <feature name="vme"/>
    </cpu>
    <power_management/>
    <migration_features>
      <live/>
      <uri_transports>
        <uri_transport>tcp</uri_transport>
      </uri_transports>
    </migration_features>
    <topology>
      <cells num="1">
        <cell id="0">
          <cpus num="8">
            <cpu id="0"/>
            <cpu id="1"/>
            <cpu id="2"/>
            <cpu id="3"/>
            <cpu id="4"/>
            <cpu id="5"/>
            <cpu id="6"/>
            <cpu id="7"/>
          </cpus>
        </cell>
      </cells>
    </topology>
    <secmodel>
      <model>apparmor</model>
      <doi>0</doi>
    </secmodel>
  </host>

  <guest>
    <os_type>hvm</os_type>
    <arch name="i686">
      <wordsize>32</wordsize>
      <emulator>/usr/bin/qemu-system-x86_64</emulator>
      <machine>pc-1.0-qemu-kvm</machine>
      <machine canonical="pc-1.0-qemu-kvm">pc-1.0-precise</machine>
      <machine canonical="pc-1.0">pc</machine>
      <machine>pc-1.0</machine>
      <machine>pc-0.14</machine>
      <machine>pc-0.13</machine>
      <machine>pc-0.12</machine>
      <machine>pc-0.11</machine>
      <machine>pc-0.10</machine>
      <machine>isapc</machine>
      <domain type="qemu">
        <emulator>/usr/bin/qemu-system-x86_64</emulator>
      </domain>
      <domain type="kvm">
        <emulator>/usr/bin/kvm</emulator>
        <machine>pc-1.0-qemu-kvm</machine>
        <machine canonical="pc-1.0-qemu-kvm">pc-1.0-precise</machine>
        <machine canonical="pc-1.0">pc</machine>
        <machine>pc-1.0</machine>
        <machine>pc-0.14</machine>
        <machine>pc-0.13</machine>
        <machine>pc-0.12</machine>
        <machine>pc-0.11</machine>
        <machine>pc-0.10</machine>
        <machine>isapc</machine>
      </domain>
    </arch>
    <features>
      <cpuselection/>
      <deviceboot/>
      <pae/>
      <nonpae/>
      <acpi default="on" toggle="yes"/>
      <apic default="on" toggle="no"/>
    </features>
  </guest>

  <guest>
    <os_type>hvm</os_type>
    <arch name="x86_64">
      <wordsize>64</wordsize>
      <emulator>/usr/bin/qemu-system-x86_64</emulator>
      <machine>pc-1.0-qemu-kvm</machine>
      <machine canonical="pc-1.0-qemu-kvm">pc-1.0-precise</machine>
      <machine canonical="pc-1.0">pc</machine>
      <machine>pc-1.0</machine>
      <machine>pc-0.14</machine>
      <machine>pc-0.13</machine>
      <machine>pc-0.12</machine>
      <machine>pc-0.11</machine>
      <machine>pc-0.10</machine>
      <machine>isapc</machine>
      <domain type="qemu">
        <emulator>/usr/bin/qemu-system-x86_64</emulator>
      </domain>
      <domain type="kvm">
        <emulator>/usr/bin/kvm</emulator>
        <machine>pc-1.0-qemu-kvm</machine>
        <machine canonical="pc-1.0-qemu-kvm">pc-1.0-precise</machine>
        <machine canonical="pc-1.0">pc</machine>
        <machine>pc-1.0</machine>
        <machine>pc-0.14</machine>
        <machine>pc-0.13</machine>
        <machine>pc-0.12</machine>
        <machine>pc-0.11</machine>
        <machine>pc-0.10</machine>
        <machine>isapc</machine>
      </domain>
    </arch>
    <features>
      <cpuselection/>
      <deviceboot/>
      <acpi default="on" toggle="yes"/>
      <apic default="on" toggle="no"/>
    </features>
  </guest>

</capabilities>

[Tue, 09 Jun 2015 09:45:48 virt-manager 12526] DEBUG (connection:754) Using domain events
[Tue, 09 Jun 2015 09:45:48 virt-manager 12526] DEBUG (connection:768) Error registering domain balloon event: internal error unsupported event ID 13
[Tue, 09 Jun 2015 09:45:48 virt-manager 12526] DEBUG (connection:768) Error registering domain tray event: internal error unsupported event ID 10
[Tue, 09 Jun 2015 09:45:48 virt-manager 12526] DEBUG (connection:768) Error registering domain device removed event: internal error unsupported event ID 15
[Tue, 09 Jun 2015 09:45:48 virt-manager 12526] DEBUG (connection:768) Error registering domain device added event: internal error unsupported event ID 19
[Tue, 09 Jun 2015 09:45:48 virt-manager 12526] DEBUG (connection:794) Error registering network events: unknown procedure: 313
[Tue, 09 Jun 2015 09:45:49 virt-manager 12526] DEBUG (connection:444) Connection doesn't seem to support interface APIs. Skipping all interface polling.
[Tue, 09 Jun 2015 09:45:51 virt-manager 12526] DEBUG (connection:1256) Error polling connection qemu+ssh://root@server/system?socket=/var/run/libvirt/libvirt-sock
Traceback (most recent call last):
  File "/usr/local/Cellar/virt-manager/1.2.0/libexec/share/virt-manager/virtManager/connection.py", line 1236, in tick_from_engine
    self._tick(*args, **kwargs)
  File "/usr/local/Cellar/virt-manager/1.2.0/libexec/share/virt-manager/virtManager/connection.py", line 1129, in _tick
    initial_poll, pollvm, pollnet, pollpool, polliface, pollnodedev)
  File "/usr/local/Cellar/virt-manager/1.2.0/libexec/share/virt-manager/virtManager/connection.py", line 1093, in _poll
    "refreshing xml for new %s" % newlist[0].class_name(),
KeyError: 0
[Tue, 09 Jun 2015 09:45:51 virt-manager 12526] DEBUG (connection:807) conn.close() uri=qemu+ssh://root@server/system?socket=/var/run/libvirt/libvirt-sock
[Tue, 09 Jun 2015 09:45:51 virt-manager 12526] DEBUG (connection:541) conn=qemu+ssh://root@server/system?socket=/var/run/libvirt/libvirt-sock changed to state=Disconnected
[Tue, 09 Jun 2015 09:45:51 virt-manager 12526] DEBUG (error:97) error dialog message:
summary=Error polling connection 'qemu+ssh://root@server/system?socket=/var/run/libvirt/libvirt-sock': 0
details=Traceback (most recent call last):
  File "/usr/local/Cellar/virt-manager/1.2.0/libexec/share/virt-manager/virtManager/engine.py", line 343, in _handle_tick_queue
    conn.tick_from_engine(**kwargs)
  File "/usr/local/Cellar/virt-manager/1.2.0/libexec/share/virt-manager/virtManager/connection.py", line 1266, in tick_from_engine
    raise e  # pylint: disable=raising-bad-type
KeyError: 0
```

**Comment by fuerst** [comment]
BTW: `virsh` to the servers works fine.

**Comment by jeffreywildman** [comment]
This may be relevant: https://bugzilla.redhat.com/show_bug.cgi?id=1227111

**Comment by fuerst** [comment]
Thanks a lot! This is a duplicate of https://bugzilla.redhat.com/show_bug.cgi?id=1219443: _virt-manager 1.2.0 is unable to list VMs on hosts with old libvirt_. The patch there does fix it. It is already commited upstream so we hopefully see it in virt-manager 1.2.1

**Comment by jeffreywildman** [comment]
Upstream patch was included in virt-manager 1.2.1, formula here updated in 3048ab3ceb7118877489eece148369e2fa59ef56
